diff --git a/arch/arm/boot/dts/include/dt-bindings b/arch/arm/boot/dts/include/dt-bindings
deleted file mode 120000
index 08c00e4..0000000
--- a/arch/arm/boot/dts/include/dt-bindings
+++ /dev/null
@@ -1 +0,0 @@
-../../../../../include/dt-bindings
\ No newline at end of file
diff --git a/arch/arm/configs/esip_defconfig b/arch/arm/configs/esip_defconfig
index f7c1afb..5d461b0 100644
--- a/arch/arm/configs/esip_defconfig
+++ b/arch/arm/configs/esip_defconfig
@@ -1296,44 +1296,7 @@ CONFIG_USB_NET_ZAURUS=m
 # CONFIG_USB_IPHETH is not set
 # CONFIG_USB_SIERRA_NET is not set
 # CONFIG_USB_VL600 is not set
-CONFIG_WLAN=y
-# CONFIG_LIBERTAS_THINFIRM is not set
-# CONFIG_AT76C50X_USB is not set
-# CONFIG_USB_ZD1201 is not set
-# CONFIG_USB_NET_RNDIS_WLAN is not set
-# CONFIG_RTL8187 is not set
-# CONFIG_MAC80211_HWSIM is not set
-CONFIG_ATH_CARDS=y
-# CONFIG_ATH_DEBUG is not set
-# CONFIG_ATH9K is not set
-# CONFIG_ATH9K_HTC is not set
-# CONFIG_CARL9170 is not set
-CONFIG_ATH6KL=m
-CONFIG_ATH6KL_SDIO=m
-# CONFIG_ATH6KL_USB is not set
-# CONFIG_ATH6KL_DEBUG is not set
-# CONFIG_AR5523 is not set
-# CONFIG_ATH10K is not set
-# CONFIG_WCN36XX is not set
-# CONFIG_B43 is not set
-# CONFIG_B43LEGACY is not set
-CONFIG_BRCMUTIL=m
-# CONFIG_BRCMSMAC is not set
-CONFIG_BRCMFMAC=m
-CONFIG_BRCMFMAC_SDIO=y
-# CONFIG_BRCMFMAC_USB is not set
-# CONFIG_BRCM_TRACING is not set
-# CONFIG_BRCMDBG is not set
-# CONFIG_HOSTAP is not set
-# CONFIG_LIBERTAS is not set
-# CONFIG_P54_COMMON is not set
-# CONFIG_RT2X00 is not set
-CONFIG_RTL_CARDS=y
-# CONFIG_RTL8192CU is not set
-# CONFIG_WL_TI is not set
-# CONFIG_ZD1211RW is not set
-# CONFIG_MWIFIEX is not set
-# CONFIG_CW1200 is not set
+# CONFIG_WLAN is not set
 
 #
 # Enable WiMAX (Networking options) to see the WiMAX drivers
@@ -1918,14 +1881,6 @@ CONFIG_WATCHDOG=y
 # USB-based Watchdog Cards
 #
 # CONFIG_USBPCWATCHDOG is not set
-CONFIG_APPWD=y
-CONFIG_OF_APPWD=y
-CONFIG_APPWD_MAX_WDT=1
-# CONFIG_APPWD_WDT_DUMMY is not set
-# CONFIG_APPWD_WDT_GPIO is not set
-# CONFIG_APPWD_WDT_MPC8xxx is not set
-CONFIG_APPWD_WDT_IMX2=y
-# CONFIG_PREMATURE_WATCHDOG is not set
 CONFIG_SSB_POSSIBLE=y
 
 #
@@ -2607,7 +2562,7 @@ CONFIG_USB=y
 #
 CONFIG_USB_DEFAULT_PERSIST=y
 # CONFIG_USB_DYNAMIC_MINORS is not set
-# CONFIG_USB_OTG is not set
+CONFIG_USB_OTG=y
 # CONFIG_USB_OTG_WHITELIST is not set
 # CONFIG_USB_OTG_BLACKLIST_HUB is not set
 # CONFIG_USB_OTG_FSM is not set
@@ -2800,43 +2755,18 @@ CONFIG_USB_GADGET_STORAGE_NUM_BUFFERS=2
 # CONFIG_USB_M66592 is not set
 # CONFIG_USB_NET2272 is not set
 # CONFIG_USB_DUMMY_HCD is not set
-CONFIG_USB_LIBCOMPOSITE=m
-CONFIG_USB_F_ACM=m
-CONFIG_USB_F_SS_LB=m
-CONFIG_USB_U_SERIAL=m
-CONFIG_USB_U_ETHER=m
-CONFIG_USB_F_SERIAL=m
-CONFIG_USB_F_OBEX=m
-CONFIG_USB_F_NCM=m
-CONFIG_USB_F_ECM=m
-CONFIG_USB_F_EEM=m
-CONFIG_USB_F_SUBSET=m
-CONFIG_USB_F_RNDIS=m
-CONFIG_USB_F_MASS_STORAGE=m
-CONFIG_USB_F_FS=m
-CONFIG_USB_CONFIGFS=m
-CONFIG_USB_CONFIGFS_SERIAL=y
-CONFIG_USB_CONFIGFS_ACM=y
-CONFIG_USB_CONFIGFS_OBEX=y
-CONFIG_USB_CONFIGFS_NCM=y
-CONFIG_USB_CONFIGFS_ECM=y
-CONFIG_USB_CONFIGFS_ECM_SUBSET=y
-CONFIG_USB_CONFIGFS_RNDIS=y
-CONFIG_USB_CONFIGFS_EEM=y
-CONFIG_USB_CONFIGFS_MASS_STORAGE=y
-CONFIG_USB_CONFIGFS_F_LB_SS=y
-CONFIG_USB_CONFIGFS_F_FS=y
-CONFIG_USB_ZERO=m
+CONFIG_USB_LIBCOMPOSITE=y
+CONFIG_USB_F_MASS_STORAGE=y
+# CONFIG_USB_CONFIGFS is not set
+# CONFIG_USB_ZERO is not set
 # CONFIG_USB_AUDIO is not set
-CONFIG_USB_ETH=m
-CONFIG_USB_ETH_RNDIS=y
-# CONFIG_USB_ETH_EEM is not set
-CONFIG_USB_G_NCM=m
-CONFIG_USB_GADGETFS=m
+# CONFIG_USB_ETH is not set
+# CONFIG_USB_G_NCM is not set
+# CONFIG_USB_GADGETFS is not set
 # CONFIG_USB_FUNCTIONFS is not set
-CONFIG_USB_MASS_STORAGE=m
-# CONFIG_FSL_UTP is not set
-CONFIG_USB_G_SERIAL=m
+CONFIG_USB_MASS_STORAGE=y
+CONFIG_FSL_UTP=y
+# CONFIG_USB_G_SERIAL is not set
 # CONFIG_USB_MIDI_GADGET is not set
 # CONFIG_USB_G_PRINTER is not set
 # CONFIG_USB_CDC_COMPOSITE is not set
@@ -2960,7 +2890,7 @@ CONFIG_LEDS_TRIGGER_GPIO=y
 # CONFIG_EDAC is not set
 CONFIG_RTC_LIB=y
 CONFIG_RTC_CLASS=y
-CONFIG_RTC_HCTOSYS=y
+# CONFIG_RTC_HCTOSYS is not set
 CONFIG_RTC_SYSTOHC=y
 CONFIG_RTC_HCTOSYS_DEVICE="rtc0"
 # CONFIG_RTC_DEBUG is not set
@@ -3085,18 +3015,12 @@ CONFIG_DMA_OF=y
 #
 CONFIG_STAGING=y
 # CONFIG_USBIP_CORE is not set
-# CONFIG_W35UND is not set
-# CONFIG_PRISM2_USB is not set
 # CONFIG_ECHO is not set
 # CONFIG_COMEDI is not set
-# CONFIG_RTLLIB is not set
-# CONFIG_R8712U is not set
-# CONFIG_R8188EU is not set
 # CONFIG_RTS5139 is not set
 # CONFIG_TRANZPORT is not set
 # CONFIG_LINE6_USB is not set
 # CONFIG_USB_SERIAL_QUATECH2 is not set
-# CONFIG_VT6656 is not set
 
 #
 # IIO staging drivers
@@ -3482,7 +3406,7 @@ CONFIG_TMPFS=y
 # CONFIG_TMPFS_POSIX_ACL is not set
 # CONFIG_TMPFS_XATTR is not set
 # CONFIG_HUGETLB_PAGE is not set
-CONFIG_CONFIGFS_FS=m
+CONFIG_CONFIGFS_FS=y
 CONFIG_MISC_FILESYSTEMS=y
 # CONFIG_ADFS_FS is not set
 # CONFIG_AFFS_FS is not set
diff --git a/arch/arm/kernel/perf_event.c b/arch/arm/kernel/perf_event.c
index 789d846..843b1c3 100644
--- a/arch/arm/kernel/perf_event.c
+++ b/arch/arm/kernel/perf_event.c
@@ -511,7 +511,7 @@ int armpmu_register(struct arm_pmu *armpmu, int type)
 {
 	armpmu_init(armpmu);
 	pm_runtime_enable(&armpmu->plat_device->dev);
-	pr_info("enabled with %s PMU driver, %d counters available\n",
+	pr_info("ENABLED with %s PMU driver, %d counters available\n",
 			armpmu->name, armpmu->num_events);
 	return perf_pmu_register(&armpmu->pmu, armpmu->name, type);
 }
diff --git a/arch/arm/kernel/setup.c b/arch/arm/kernel/setup.c
index 1e8b030..801449d 100644
--- a/arch/arm/kernel/setup.c
+++ b/arch/arm/kernel/setup.c
@@ -484,7 +484,7 @@ void __init smp_setup_processor_id(void)
 	 */
 	set_my_cpu_offset(0);
 
-	pr_info("Booting Linux on physical CPU 0x%x\n", mpidr);
+	pr_info("ESIP--UTP--Booting Linux on physical CPU 0x%x\n", mpidr);
 }
 
 struct mpidr_hash mpidr_hash;
diff --git a/arch/metag/boot/dts/include/dt-bindings b/arch/metag/boot/dts/include/dt-bindings
deleted file mode 120000
index 08c00e4..0000000
--- a/arch/metag/boot/dts/include/dt-bindings
+++ /dev/null
@@ -1 +0,0 @@
-../../../../../include/dt-bindings
\ No newline at end of file
diff --git a/arch/mips/boot/dts/include/dt-bindings b/arch/mips/boot/dts/include/dt-bindings
deleted file mode 120000
index 08c00e4..0000000
--- a/arch/mips/boot/dts/include/dt-bindings
+++ /dev/null
@@ -1 +0,0 @@
-../../../../../include/dt-bindings
\ No newline at end of file
diff --git a/arch/powerpc/boot/dts/include/dt-bindings b/arch/powerpc/boot/dts/include/dt-bindings
deleted file mode 120000
index 08c00e4..0000000
--- a/arch/powerpc/boot/dts/include/dt-bindings
+++ /dev/null
@@ -1 +0,0 @@
-../../../../../include/dt-bindings
\ No newline at end of file
diff --git a/drivers/rtc/hctosys.c b/drivers/rtc/hctosys.c
index 4aa60d7..623cddc 100644
--- a/drivers/rtc/hctosys.c
+++ b/drivers/rtc/hctosys.c
@@ -57,20 +57,21 @@ static int __init rtc_hctosys(void)
 	err = do_settimeofday(&tv);
 
 	dev_info(rtc->dev.parent,
-		"setting system clock to "
+		"ESIP--OTG--setting system clock to "
 		"%d-%02d-%02d %02d:%02d:%02d UTC (%u)\n",
 		tm.tm_year + 1900, tm.tm_mon + 1, tm.tm_mday,
 		tm.tm_hour, tm.tm_min, tm.tm_sec,
 		(unsigned int) tv.tv_sec);
-
+		printk("%s ZS:TP1\n",__func__);
+		printk("%s ZS:TP0\n",__func__);printk("%s ZS:TP0\n",__func__);printk("%s ZS:TP0\n",__func__);printk("%s ZS:TP0\n",__func__);printk("%s ZS:TP0\n",__func__);printk("%s ZS:TP0\n",__func__);
 err_invalid:
-err_read:
-	rtc_class_close(rtc);
-
-err_open:
-	rtc_hctosys_ret = err;
+printk("%s ZS1:TP1\n",__func__);
+err_read: printk("%s ZS1:TP2\n",__func__);
+	rtc_class_close(rtc); printk("%s ZS1:TP3\n",__func__);
 
+err_open:printk("%s ZS1:TP4\n",__func__);
+	rtc_hctosys_ret = err;printk("%s ZS1:TP5\n",__func__);
+	printk("%s ZS1:TP6\n",__func__);
 	return err;
 }
-
 late_initcall(rtc_hctosys);
diff --git a/drivers/rtc/interface.c b/drivers/rtc/interface.c
index 544be72..209c4d6 100644
--- a/drivers/rtc/interface.c
+++ b/drivers/rtc/interface.c
@@ -620,8 +620,11 @@ EXPORT_SYMBOL_GPL(rtc_class_open);
 
 void rtc_class_close(struct rtc_device *rtc)
 {
+	printk("%s ZS2:TP0\n",__func__);	
 	module_put(rtc->owner);
+	printk("%s ZS2:TP0\n",__func__);
 	put_device(&rtc->dev);
+	printk("%s ZS2:TP0\n",__func__);
 }
 EXPORT_SYMBOL_GPL(rtc_class_close);
 
diff --git a/drivers/staging/imx-drm/Kconfig b/drivers/staging/imx-drm/Kconfig
index 78319ad..8604a97 100644
--- a/drivers/staging/imx-drm/Kconfig
+++ b/drivers/staging/imx-drm/Kconfig
@@ -1,11 +1,11 @@
 config DRM_IMX
 	tristate "DRM Support for Freescale i.MX"
-	select DRM_KMS_HELPER
-	select DRM_KMS_FB_HELPER
-	select VIDEOMODE_HELPERS
-	select DRM_GEM_CMA_HELPER
-	select DRM_KMS_CMA_HELPER
-	depends on DRM && (ARCH_MXC || ARCH_MULTIPLATFORM)
+#	select DRM_KMS_HELPER
+#	select DRM_KMS_FB_HELPER
+#	select VIDEOMODE_HELPERS
+#	select DRM_GEM_CMA_HELPER
+#	select DRM_KMS_CMA_HELPER
+#	depends on DRM && (ARCH_MXC || ARCH_MULTIPLATFORM)
 	help
 	  enable i.MX graphics support
 
diff --git a/drivers/video/mxc/mxc_hdmi.c b/drivers/video/mxc/mxc_hdmi.c
index 10b54fb..f15de6d 100644
--- a/drivers/video/mxc/mxc_hdmi.c
+++ b/drivers/video/mxc/mxc_hdmi.c
@@ -2762,6 +2762,8 @@ static int mxc_hdmi_probe(struct platform_device *pdev)
 	struct resource *res;
 	int ret = 0;
 
+	printk ("ESIP board --ZS\r\n");
+
 	/* Check I2C driver is loaded and available
 	 * check hdcp function is enable by dts */
 	hdmi_hdcp_get_property(pdev);
diff --git a/fs/ext4/super.c b/fs/ext4/super.c
index 9fb3e6c..253d7e1 100644
--- a/fs/ext4/super.c
+++ b/fs/ext4/super.c
@@ -3375,7 +3375,7 @@ static int ext4_fill_super(struct super_block *sb, void *data, int silent)
 	int err = 0;
 	unsigned int journal_ioprio = DEFAULT_JOURNAL_IOPRIO;
 	ext4_group_t first_not_zeroed;
-
+	printk("%s ZS7:TP1.",__func__);
 	sbi = kzalloc(sizeof(*sbi), GFP_KERNEL);
 	if (!sbi)
 		goto out_free_orig;
diff --git a/init/do_mounts.c b/init/do_mounts.c
index 8e5addc..1d42956 100644
--- a/init/do_mounts.c
+++ b/init/do_mounts.c
@@ -358,6 +358,7 @@ static void __init get_fs_names(char *page)
 
 static int __init do_mount_root(char *name, char *fs, int flags, void *data)
 {
+	printk("%sZS7:TP0\n",__func__);
 	struct super_block *s;
 	int err = sys_mount(name, "/root", fs, flags, data);
 	if (err)
@@ -376,6 +377,7 @@ static int __init do_mount_root(char *name, char *fs, int flags, void *data)
 
 void __init mount_block_root(char *name, int flags)
 {
+	printk("%s ZS7:TP3\n",__func__);
 	struct page *page = alloc_page(GFP_KERNEL |
 					__GFP_NOTRACK_FALSE_POSITIVE);
 	char *fs_names = page_address(page);
diff --git a/init/main.c b/init/main.c
index 58c132d..5f96567 100644
--- a/init/main.c
+++ b/init/main.c
@@ -910,23 +910,25 @@ static noinline void __init kernel_init_freeable(void)
 
 	smp_init();
 	sched_init_smp();
-
+	printk("%s ZS:TP4\n",__func__);
 	do_basic_setup();
-
+	printk("%s ZS:TP3\n",__func__);
+	printk("%d ZS8:TP000\n",ramdisk_execute_command);
 	/* Open the /dev/console on the rootfs, this should never fail */
 	if (sys_open((const char __user *) "/dev/console", O_RDWR, 0) < 0)
-		pr_err("Warning: unable to open an initial console.\n");
-
+		pr_err("Warning: unable to open an initial console.\n"); 
+	printk("%s ZS8:TP2\n",__func__);
 	(void) sys_dup(0);
 	(void) sys_dup(0);
 	/*
 	 * check if there is an early userspace init.  If yes, let it do all
 	 * the work
 	 */
-
+	printk("%s ZS8:TP3\n",__func__);
+	printk("%d ZS8:TP00\n",ramdisk_execute_command);
 	if (!ramdisk_execute_command)
 		ramdisk_execute_command = "/init";
-
+	printk("%s ZS8:TP4\n",__func__);
 	if (sys_access((const char __user *) ramdisk_execute_command, 0) != 0) {
 		ramdisk_execute_command = NULL;
 		prepare_namespace();
@@ -939,5 +941,6 @@ static noinline void __init kernel_init_freeable(void)
 	 */
 
 	/* rootfs is available now, try loading default modules */
+	printk("%s ZS8:TP1\n",__func__);
 	load_default_modules();
 }
diff --git a/sound/last.c b/sound/last.c
index 43f2228..87eab3e 100644
--- a/sound/last.c
+++ b/sound/last.c
@@ -27,6 +27,7 @@ static int __init alsa_sound_last_init(void)
 	int idx, ok = 0;
 	
 	printk(KERN_INFO "ALSA device list:\n");
+	printk("%s ZS:TP2 \n",__func__);
 	for (idx = 0; idx < SNDRV_CARDS; idx++)
 		if (snd_cards[idx] != NULL) {
 			printk(KERN_INFO "  #%i: %s\n", idx, snd_cards[idx]->longname);
